### PR TITLE
[Samples][Java] Fix disparities of 14.nlp-with-dispatch

### PR DESCRIPTION
**Pull request by VictorGrycuk**
Fixes #

## Proposed Changes

  - 
  -
  -

### CODE REVIEW

File: samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java [hunk 1]
```diff
@@ -28,25 +25,24 @@
...
+        this.botServices = botServices;
     }
 
     @Override
     protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
-        // First, we use the dispatch model to determine which cognitive service (LUS or
-        // QnA) to use.
-        RecognizerResult recognizerResult = _botServices.getDispatch().recognize(turnContext).join();
+        // First, we use the dispatch model to determine which cognitive service (LUIS or QnA) to use.
+        return botServices.getDispatch().recognize(turnContext).thenCompose(result -> {
```
Comment (Batta32) [inline samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java]:
```suggestion
        return botServices.getDispatch().recognize(turnContext).thenCompose(recognizerResult -> {
```

Before was `recognizerResult`

File: samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java [hunk 2]
```diff
@@ -28,25 +25,24 @@
...
+            // Top intent tell us which cognitive service to use.
+            NamedIntentScore topIntent = result.getTopScoringIntent();
```
Comment (Batta32) [inline samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java]:
```suggestion
            NamedIntentScore topIntent = recognizerResult.getTopScoringIntent();
```

File: samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java [hunk 3]
```diff
@@ -28,25 +25,24 @@
...
 
-        // Top intent tell us which cognitive service to use.
-        NamedIntentScore topIntent = recognizerResult.getTopScoringIntent();
-
-        // Next, we call the dispatcher with the top intent.
-        return dispatchToTopIntent(turnContext, topIntent.intent, recognizerResult);
+            // Next, we call the dispatcher with the top intent.
+            return dispatchToTopIntent(turnContext, topIntent.intent, result).thenApply(task -> null);
```
Comment (Batta32) [inline samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java]:
```suggestion
            return dispatchToTopIntent(turnContext, topIntent.intent, recognizerResult).thenApply(task -> null);
```

File: samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java [hunk 4]
```diff
@@ -17,9 +17,6 @@
 import com.microsoft.bot.builder.RecognizerResult;
 import com.microsoft.bot.builder.TurnContext;
 import com.microsoft.bot.builder.RecognizerResult.NamedIntentScore;
-import com.microsoft.bot.sample.nlpwithdispatch.BotServices;
-import com.microsoft.bot.sample.nlpwithdispatch.Intent;
-import com.microsoft.bot.sample.nlpwithdispatch.PredictionResult;
```
Comment (Batta32) [inline samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java:22]: Is it correct to remove these? They are used in this file 🤔

**Review by Batta32 — CHANGES_REQUESTED** [review]
About the README:
1. We are missing the [Overview](https://github.com/microsoft/BotBuilder-Samples/tree/main/Migration/Dispatch/csharp/14.nlp-with-dispatch#overview) section
2. We are missing the `Language Understanding using LUIS` link (see [here](https://github.com/microsoft/BotBuilder-Samples/tree/main/Migration/Dispatch/csharp/14.nlp-with-dispatch#further-reading))

File: samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java (hunk 4 above)
Comment (VictorGrycuk) [reply samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java:22]: That's weird, I didn't manually remove them. No idea what happened.
However, the sample did work as it was, so it was either after testing it or they are not actually used.
I will validate this and revert it if necessary.

Comment (VictorGrycuk) [reply samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java:22]: Are you sure we need those imports? All those classes are inside the same package, so they are within the scope and reachable

Comment (VictorGrycuk) [reply samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java:22]: Now I remember what happened. The bot class was in a different folder, therefore needed those imports. After it was moved to the root folder, those imports were not needed anymore.

**Pull request closed without merging**